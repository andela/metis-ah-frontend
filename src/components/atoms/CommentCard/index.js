import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import timePast from 'time_past';
import _ from 'underscore';

const CommentCard = (props) => {
  const { comment } = props;
  const name = `${comment.user.firstname} ${comment.user.lastname}`;
  const userImage = 'https://res.cloudinary.com/dbsxxymfz/image/upload/v1536757459/dummy-profile.png';
  return (
    <Fragment>
      <div id="comments-view">
        <div className="comment-body">
          {_.unescape(comment.content)}
        </div>
        <div className="comment-footer">
          <img src={comment.user.image || userImage} alt="userImage" />
          <p className="userName">
            <Link to={`/users/${comment.user.id}`}>{name.trim() ? name : comment.user.username}</Link>
          </p>
          <p className="time-ago">
            {timePast(comment.createdAt)}
          </p>
        </div>
      </div>
    </Fragment>
  );
};

export default CommentCard;
