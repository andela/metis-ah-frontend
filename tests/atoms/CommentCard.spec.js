import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore();

const mockFunction = param => jest.fn(param);

import CommentCard from '../../src/components/atoms/CommentCard';

const props = {
  comment: {
    id: 9,
    content: 'Thanks',
    edited: false,
    createdAt: '2018-11-13T16:13:29.901Z',
    updatedAt: '2018-11-13T16:13:29.901Z',
    articleId: 170,
    userId: 11,
    user: {
      id: 11,
      username: 'johngorithm',
      firstname: 'JAY',
      lastname: 'OBI',
      image:
        'http://res.cloudinary.com/dbsxxymfz/image/upload/v1541429992/es8jfxtycdfjvec6ha1a.png',
    },
    likes: [],
    likeCount: 0,
  },
};

const comment = {
  id: 9,
  content: 'Thanks',
  edited: false,
  createdAt: '2018-11-13T16:13:29.901Z',
  updatedAt: '2018-11-13T16:13:29.901Z',
  articleId: 170,
  userId: 11,
  user: {
    id: 11,
    username: 'johngorithm',
    firstname: undefined,
    lastname: undefined,
    image: undefined,
  },
  likes: [],
  likeCount: 0,
};

describe('CommentCard Tests', () => {
  test('it renders successfully', () => {
    const wrapper = shallow(
      <Provider store={store} handleLike={mockFunction}>
        <CommentCard {...props} />
      </Provider>
    );
    expect(wrapper.exists()).toBe(true);
  });

  test('it matches snapshot', () => {
    const wrapper = shallow(
      <Provider store={store} handleLike={mockFunction}>
        <CommentCard {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('it renders username when firstname and lastname are undefined', () => {
    const wrapper = shallow(
      <Provider store={store} handleLike={mockFunction}>
        <CommentCard comment={comment} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
