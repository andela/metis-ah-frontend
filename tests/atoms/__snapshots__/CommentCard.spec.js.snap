// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard Tests it matches snapshot 1`] = `
<Fragment>
  <div
    id="comments-view"
  >
    <div
      className="comment-body"
    >
      Thanks
    </div>
    <div
      className="comment-footer"
    >
      <img
        alt="userImage"
        src="http://res.cloudinary.com/dbsxxymfz/image/upload/v1541429992/es8jfxtycdfjvec6ha1a.png"
      />
      <p
        className="userName"
      >
        <Link
          replace={false}
          to="/users/11"
        >
          JAY OBI
        </Link>
      </p>
      <p
        className="time-ago"
      >
        5 days ago
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`CommentCard Tests it renders username when firstname and lastname are undefined 1`] = `
<Fragment>
  <div
    id="comments-view"
  >
    <div
      className="comment-body"
    >
      Thanks
    </div>
    <div
      className="comment-footer"
    >
      <img
        alt="userImage"
        src="https://res.cloudinary.com/dbsxxymfz/image/upload/v1536757459/dummy-profile.png"
      />
      <p
        className="userName"
      >
        <Link
          replace={false}
          to="/users/11"
        >
          undefined undefined
        </Link>
      </p>
      <p
        className="time-ago"
      >
        5 days ago
      </p>
    </div>
  </div>
</Fragment>
`;
